app/models/subscription.rb

```diff
@@ -0,0 +1,21 @@
+class Subscription < ApplicationRecord
+  belongs_to :user
+  belongs_to :plan
+
+  has_many :transactions
+
+  validates :billing_day, presence: true, numericality: { greater_than: 0, less_than_or_equal_to: 28 }
+
+  before_validation :set_billing_day
+  after_save :subscription_transaction
+
+  private
+
+  def set_billing_day
+    self.billing_day = [Date.today.day, 28].min
```

Comment:
self is not advised to be used inside the model unless you have another variable of same name.